surefire-report-parser/src/main/java/org/apache/maven/plugins/surefire/report/TestSuiteXmlParser.java

```diff
@@ -196,7 +196,8 @@ else if ( "failure".equals( qName ) )
                 }
                 else if ( "error".equals( qName ) )
                 {
-                    testCase.setFailure( attributes.getValue( "message" ), attributes.getValue( "type" ) );
+                    testCase.setFailure( attributes.getValue( "message" ), attributes.getValue( "type" ) )
```

Comment:
If having an error, then failure and error are set both. Why?
See Line 235. Maybe you should introduce `setErrorDetail` similar to what is `setFailureDetail`, and show the detail in HTML report and thus distinguish between these two.

Reply:
> If having an error, then failure and error are set both. Why?

An error is a specific kind of failure: if I start to introduce error details, then I should also add error type, message, line, and I'll got every field of `ReportTestCase` duplicated. My purpose was to keep the field of `ReportTestCase` to report the information around the failure, but to be able to flag it as an error instead of a failure.